klippy/extras/exclude_object.py

```diff
@@ -0,0 +1,263 @@
+            "current_object": self.current_object
+        }
+        return status
+
+    def move(self, newpos, speed):
+        move_in_excluded_region = self._test_in_excluded_region()
+        self.last_speed = speed
+
+        if move_in_excluded_region:
+            if self.in_excluded_region:
+                self._ignore_move(newpos, speed)
+            else:
+                self._move_into_excluded_region(newpos, speed)
+        else:
+            if self.in_excluded_region:
+                self._move_from_excluded_region(newpos, speed)
+            else:
+                self._normal_move(newpos, speed)
+
+    cmd_START_CURRENT_OBJECT_help = "Marks the beginning the current object" \
+                                    " as labeled"
+    def cmd_START_CURRENT_OBJECT(self, params):
+        name = params.get('NAME').upper()
+        self.current_object = name
+        self.was_excluded_at_start = self._test_in_excluded_region()
+    cmd_END_CURRENT_OBJECT_help = "Markes the end the current object"
+    def cmd_END_CURRENT_OBJECT(self, gcmd):
+        if self.current_object == None and self.next_transform:
+            gcmd.respond_info("END_CURRENT_OBJECT called, but no object is"
+                              " currently active")
+            return
+        name = gcmd.get('NAME', default=None)
+        if name != None and name.upper() != self.current_object:
+            gcmd.respond_info("END_CURRENT_OBJECT NAME=%s does not match the"
+                              " current object NAME=%s" %
+                              (name.upper(), self.current_object))
+
+        is_excluded = self._test_in_excluded_region()
+        self.current_object = None
+    cmd_EXCLUDE_OBJECT_help = "Cancel moves inside a specified objects"
+    def cmd_EXCLUDE_OBJECT(self, params):
+        name = params.get('NAME').upper()
+        if name not in self.excluded_objects:
+            self.excluded_objects.append(name)
+    cmd_EXCLUDE_OBJECT_RESET_help = "Resets the exclude_object state by" \
+                                    " clearing the list of object definitions" \
+                                    " and removed objects"
+    def cmd_EXCLUDE_OBJECT_RESET(self, params):
+        self._reset_file()
+    cmd_LIST_OBJECTS_help = "Lists the known objects"
+    def cmd_LIST_OBJECTS(self, gcmd):
+        if gcmd.get('VERBOSE', None) is not None:
+            object_list = " ".join (str(x) for x in self.objects.values())
+        else:
+            object_list = " ".join(self.objects.keys())
+        gcmd.respond_info(object_list)
+    cmd_LIST_EXCLUDED_OBJECTS_help = "Lists the excluded objects"
+    def cmd_LIST_EXCLUDED_OBJECTS(self, gcmd):
+        object_list = " ".join (str(x) for x in self.excluded_objects)
+        gcmd.respond_info(object_list)
+    cmd_DEFINE_OBJECT_help = "Provides a summary of an object"
+    def cmd_DEFINE_OBJECT(self, params):
+        self._setup_transform()
```

Comment:
The docs indicate that one does not need to "define objects", but as this is the only place the transform is registered, it would seem one must define at least one object.

Reply:
I'll likely change to to load when an `EXCLUDE_OBJECT` command is received.